### ISSUE TITLE
Step Function asl.json error for Intrinsic functions

### ISSUE THREAD

**Issue by KieranFleckney** [aws/aws-toolkit-vscode#3049]
## Problem
Error message displayed for using a Intrinsic functions even when valid. Error message `The value for the field must be a valid JSONPath or intrinsic function expression.` displays.
![image](https://user-images.githubusercontent.com/57374898/207205402-410afb68-308e-4a12-93b5-9f9144ad39ac.png)

## Steps to reproduce the issue

1. Create new state that uses a `Intrinsic functions`
```
        "Generate ID": {
            "Type": "Pass",
            "Next": "Merge Id",
            "Parameters": {
                "original.$": "$",
                "new": {
                    "body": {
                        "Id.$": "States.UUID()"
                    }
                }
            }
        },
```

## Expected behavior
I would expect not to get this error. This is valid on the AWS console and works fine when ran.

## System details (run the `AWS: About Toolkit` command)

-   OS: Window 11
-   Visual Studio Code version: 1.74.0
-   AWS Toolkit version: 1.58.0

**Comment by Slupov** [comment]
Same here. System details:
OS: Darwin arm64 21.6.0
Visual Studio Code extension host:  1.74.1
AWS Toolkit:  1.58.0
node: 16.14.2
electron: 19.1.8

**Comment by weisisheng** [comment]
Same here. System details:
OS: Ubuntu x86-64 22.04
AWS Toolkit: 1.61.0
node: 16.18.1

**Comment by viktor-stravito** [comment]
Temporary workaround to still be able to use uuids: `States.Format('{}', States.UUID())`.

**Comment by justinmk3** [comment]
Fixed in AWS Toolkit 1.63
